middleware/logging.go

```diff
@@ -24,6 +24,12 @@ func (l Log) logWithRequest(r *http.Request) logging.Interface {
 		l.Log = l.Log.WithField("traceID", traceID)
 	}
 
+	// Add the source IPs derived from X-Forwarded-For and the request remote address
+	sourceIPs := GetSource(r)
+	if sourceIPs != "" {
+		l.Log = l.Log.WithField("sourceIPs", sourceIPs)
```

Comment:
Assigning to a member of a pass-by-value parameter strikes me as a bit weird; I see you didn't introduce it but I would prefer you changed this function to use a regular local variable for this purpose.

Reply:
Using a local variable now.